solver/src/driver.rs

```diff
@@ -195,10 +195,33 @@ impl Driver {
             );
             // This is an additional debug log so that the log message doesn't get too long as
             // settlement information is recoverable through tenderly anyway.
-            tracing::warn!("settlement failure for: \n{:#?}", settlement.settlement,);
+            tracing::warn!("settlement failure for: \n{:#?}", settlement,);
         }
     }
 
+    // Rate settlements, ignoring those for which the rating procedure failed.
+    async fn rate_settlements(
+        &self,
+        settlements: Vec<Settlement>,
+        prices: &HashMap<H160, BigRational>,
+    ) -> Result<Vec<RatedSettlement>> {
+        let mut result: Vec<RatedSettlement> = Vec::new();
+        for settlement in settlements {
+            let surplus = settlement.total_surplus(prices);
+            let gas_estimate = settlement_submission::estimate_gas(
+                &self.settlement_contract,
+                &settlement.clone().into(),
+            )
+            .await?;
```

Comment:
The way this is written if only one of the gas estimates fails we fail rating all of them. While I think it's unlikely that a single one would fail maybe we should be more robust and error log while continuing with the rest.
Doing this is related to the batch transport suggestion too.

Reply:
If one of the gas estimates fail then the corresponding settlement will have "undefined" objective value. What to do about the "max" part then? Should we ignore the ones with undefined gas, or abort all together?

Reply:
I'd vote for skipping the ones for which gas cannot be estimated. This could happen e.g. if a new block was mined and now the simulation no longer works due to a race condition. In this case we'd also fail to estimate gas.

Reply:
Done. Let me know if it's ok now, as I used some new tricks I just learned.